### DISCUSSION TITLE
How do I calculate percentage in c++ ?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#52667] [Programming Help] [answered]
### Body

Sometimes my code works like 10/10 = 100%, but if the two numbers aren’t equal it always says 0.
My code is here:
https://github.com/PyTerminal/MarkCalculator/blob/main/main.cpp
### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment, marked as answer]
I think this is happening because you are using the int data type, try using float related data types like float or double for your variable, this should solve the issue you are having.

:)

**Reply by ghost** [reply]
Thanks! That actually helped.

**Comment by opegamanalitarok** [comment]
In C++, if you divide two integers, the result will be truncated to an integer. For example, if you divide 9 by 10, the result will be 0, not 0.9. This is why your code is not calculating the percentage correctly when the two numbers are not equal.

To fix this issue, you need to convert one or both of the numbers to a floating-point type before performing the division. One way to do this is to cast one of the integers to a double or float type. Here's an example code snippet that should work:

#include <iostream>
using namespace std;

int main() {
   int num1 = 8, num2 = 10;
   double percentage = (double)num1 / num2 * 100;
   cout << "Percentage: " << percentage << "%" << endl;
   return 0;
}

In this code, we cast the first integer num1 to a double type before dividing it by num2. This ensures that the division is performed using floating-point arithmetic, which gives us the correct result. Finally, we multiply the result by 100 to get the percentage and output it to the console.
